code/game/objects/items/stacks/sheets/silk.dm

```diff
@@ -0,0 +1,140 @@
+
+/obj/item/stack/sheet/silk/green_advanced
+	name = "Advanced Green Silk"
+	desc = "Silk woven from a gun bot... Can be used to upgrade your armor. Looks like it is from the advanced variety of silk."
+	added_armor = new(red = 10, black = -10)
+	merge_type = /obj/item/stack/sheet/silk/green_advanced
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "advanced_green_silk"
+
+/obj/item/stack/sheet/silk/steel_simple
+	name = "Simple Steel Silk"
+	desc = "Silk woven from a gene corp remnant... Can be used to upgrade your armor. Looks like it is from the simple variety of silk."
+	added_armor = new(red = 5, white = -5)
+	merge_type = /obj/item/stack/sheet/silk/steel_simple
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "simple_steel_silk"
+
+/obj/item/stack/sheet/silk/amber_simple
+	name = "Simple Amber Silk"
+	desc = "Silk woven from a carnivores worm... Can be used to upgrade your armor. Looks like it is from the simple variety of silk."
+	added_armor = new(red = -5, black = 5)
+	merge_type = /obj/item/stack/sheet/silk/amber_simple
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "simple_amber_silk"
+
+/obj/item/stack/sheet/silk/human_simple
+	name = "Simple Human Silk"
+	desc = "Silk woven from a... Human? How horrific... Can be used to upgrade your armor. Looks like it is from the simple variety of silk."
+	added_armor = new(pale = 5, white = -2.5, red = -2.5)
+	merge_type = /obj/item/stack/sheet/silk/human_simple
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "simple_human_silk"
+
+/obj/item/stack/sheet/silk/human_advanced
+	name = "Advanced Human Silk"
+	desc = "Silk woven from a... Human? How horrific... Can be used to upgrade your armor. Looks like it is from the advanced variety of silk."
+	added_armor = new(pale = 10, white = -5, red = -5)
+	merge_type = /obj/item/stack/sheet/silk/human_advanced
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "advanced_human_silk"
+
+/obj/item/stack/sheet/silk/human_elegant
+	name = "Elegant Human Silk"
+	desc = "Silk woven from a... Human? How horrific... Can be used to upgrade your armor. Looks like it is from the elegant variety of silk."
+	added_armor = new(pale = 15, white = -7.5, red = -7.5)
+	merge_type = /obj/item/stack/sheet/silk/human_elegant
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "elegant_human_silk"
+
+/obj/item/stack/sheet/silk/human_masterpiece
+	name = "Masterpiece Human Silk"
+	desc = "Silk woven from a... Human? How horrific... Can be used to upgrade your armor. Looks like it is from the masterpiece variety of silk. The best of the best."
+	added_armor = new(pale = 20, white = -10, red = -10)
+	merge_type = /obj/item/stack/sheet/silk/human_masterpiece
+	icon = 'icons/obj/stack_objects.dmi'
+	icon_state = "masterpiece_human_silk"
+
+/datum/component/butchering/silkbutchering
+
+/datum/component/butchering/silkbutchering/checkButchering(obj/item/source, mob/living/M, mob/living/user)
+	return istype(source, /obj/item/silkknife) && M.silk_results
```

Comment:
```suggestion
/datum/component/butchering/silkbutchering/checkButchering(obj/item/source, mob/living/M, mob/living/user)
	return M.silk_results
```
Source check is redundant here and serves no purpose outside of making sure this component can only be on one item which is kinda lame and not what components are for.